src/app-service.ts

```diff
@@ -271,6 +284,12 @@ export class AppService extends EventEmitter {
                 this.emit("type:" + event.type, event);
             }
         }
+        for (const event of ephemeral) {
+            this.emit("ephemeral", event);
```

Comment:
For loop of user content.

Is `.emit()` synchronously executed?
Do we need to worry about the potential length of this array? It is user input.
This could become a way to stun the bridge, if it takes a long time.

Reply:
Here we trust that the homeserver is doing due diligence of rate limiting users and/or de-duplicating things like rapid typing.

Reply:
Resolving as not a concern for now.